smpplib/command.py

```diff
@@ -177,11 +177,19 @@ def _generate_string(self, field):
                 field_value = field_value[0:self.params[field].max - 1]
 
             if field_value:
-                value = field_value + chr(0)
+                if not isinstance(field_value, bytes):
+                    value = field_value + chr(0)
+                else:
+                    value = field_value + chr(0).encode()
             else:
-                value = chr(0)
+                if not isinstance(field_value, bytes):
+                    value = chr(0)
+                else:
+                    value = chr(0).encode()
 
         setattr(self, field, field_value)
+        if isinstance(value, bytes):
```

Comment:
```suggestion
        if isinstance(value, six.binary_type):
```

hm, this check is required though

Reply:
Probably right :) You are trying to keep compatibility with 2.7

Reply:
for the test that was happening for me for every received packet, as I was testing the generated PDU equals to the received.

pdu = self.read_pdu(file)
gen_bin = pdu.generate()

And so, the exception occures. Seems nobody had long not using that part, as I was fixing that over a year ago for myself and realized the bug is still there just yesterday after checking in a new version :)

Reply:
can you please add a new function or a new file with a new function to the `tests/` directory?